### ISSUE TITLE
Web - Missing Default Merchant Statement in Split Bill Details

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#25885]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Click on the plus sign and select "Split Bill."
2. Enter the desired numbers and add an email address.
3. Click on "See More" and observe the not-updated default statement
4. Open the split bill details and click on "See More" again.
5. Notice that the default merchant statement is not present.

## Expected Result:
The default merchant statement should be present both in the split bill details and when splitting the bill.

## Actual Result:
The default merchant statement is present when splitting the bill but is not displayed in the split bill details.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.57-2

**Reproducible in staging?:** Y

**Reproducible in production?:** N
 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/891a6b5a-ef2b-425d-b00f-043bccff7509

https://github.com/Expensify/App/assets/115492554/ed37e909-73a4-440a-b849-ac03b0e73993

**Expensify/Expensify Issue URL:**

**Issue reported by:** @TewodrosGirmaA

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692855560586859

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alphaboss1104** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.

Missing Default Merchant Statement in Split Bill Details

### What is the root cause of that problem?

We displayed the merchant statement of split bill details here:

https://github.com/Expensify/App/blob/b38181a2d567cc3cbb5b426f0585db4286747fa6/src/components/MoneyRequestConfirmationList.js#L399-L407

But we didn't passed the `props.iouMerchant` from the parent.

https://github.com/Expensify/App/blob/b38181a2d567cc3cbb5b426f0585db4286747fa6/src/pages/iou/SplitBillDetailsPage.js#L85-L97

This is the root cause of this issue.

### What changes do you think we should make in order to solve the problem?

First of all we should get the merchant statement of split bill details [here](https://github.com/Expensify/App/blob/b38181a2d567cc3cbb5b426f0585db4286747fa6/src/pages/iou/SplitBillDetailsPage.js#L75):

```js
const {amount: splitAmount, currency: splitCurrency, comment: splitComment, merchant: splitMerchant} = ReportUtils.getTransactionDetails(transaction);
```
and then We should add `iouMerchant` as a prop to `MoneyRequestConfirmationList` component [here](https://github.com/Expensify/App/blob/b38181a2d567cc3cbb5b426f0585db4286747fa6/src/pages/iou/SplitBillDetailsPage.js#L86):

```js
                  <MoneyRequestConfirmationList
                      hasMultipleParticipants
                      payeePersonalDetails={payeePersonalDetails}
                      selectedParticipants={participantsExcludingPayee}
                      iouAmount={splitAmount}
                      iouComment={splitComment}
+                     iouMerchant={splitMerchant}
                      iouCurrencyCode={splitCurrency}
                      iouType={CONST.IOU.MONEY_REQUEST_TYPE.SPLIT}
                      isReadOnly
                      shouldShowFooter={false}
                  />
```

### What alternative solutions did you explore? (Optional)

**Comment by pecanoro** [comment]
Trying to figure out from which PR is coming

**Comment by pecanoro** [comment]
@mountiny @luacmartins This deploy blocker is coming from this PR https://github.com/Expensify/App/pull/25794

**Comment by luacmartins** [comment]
I don't think this is a blocker. It's a really small issue and we're still making several changes to this flow.

**Comment by pecanoro** [comment]
@situchan Another friendly bump!

**Comment by situchan** [comment]
reviewing in a few hrs

**Comment by alphaboss1104** [comment]
Question 1: It seems to me that it is expected behaviour for merchant to be inconsistent between the two pages.
                   And Should we display the email or displayName  if the user has the displayName?
Question 2: In my opinion, if the merchant is long, display the merchant in single line is better.

**Comment by pecanoro** [comment]
> merchant is inconsistent between 2 pages. Is it expected?
What do you mean by this? Different types of requests?

> if merchant is long, fully display in multiline or current behavior is fine which is single line?
Yeah, one line is fine

**Comment by situchan** [comment]
> What do you mean by this? Different types of requests?

Before split: `Request`
After split: `Bill split with xxx, xxx`

**Comment by pecanoro** [comment]
>Before split: Request
>After split: Bill split with xxx, xxx

@mountiny @luacmartins Is this expected? I know we are still refactoring some of this stuff so I am not sure if it's expected or not.

**Comment by mountiny** [comment]
I would probably wait with the split bill after the conference, I dont think this was properly scoped in terms of how the merchant and fields should behave for split so this should be brought to slack first

**Comment by pecanoro** [comment]
Gotcha! Going to put a HOLD then and move it to weekly

**Comment by pecanoro** [comment]
Bringing the discussion to Slack

**Comment by pecanoro** [comment]
Still HOLDING

**Comment by pecanoro** [comment]
Still HOLDING
